File: leetcode/215_kth_largest_element.py
import heapq

def kth_largest_element(nums, k):
    min_heap = []
    for num in nums:
        heapq.heappush(min_heap, num)
        if len(min_heap) == k + 1:
            heapq.heappop(min_heap)
    print(min_heap)
    return heapq.heappop(min_heap)

input = [3,2,1,5,6,4]
print(kth_largest_element(input, 3))

File: leetcode/_215_kth_largest_element.java
package leetcode;

import java.util.Arrays;
import java.util.PriorityQueue;

public class _215_kth_largest_element {
    public static void main(String[] args) {
        int[] nums = {3,2,3,1,2,4,5,5,6};
        int k = 3;

        System.out.println(new _215_kth_largest_element().findKthLargest(nums, k));
    }

    public int findKthLargest(int[] nums, int k) {
        var minHeap = new PriorityQueue<Integer>();

        for(int num: nums) {
            minHeap.add(num);
            if(minHeap.size() > k) {
                minHeap.poll();
            }
        }

        return minHeap.poll();
    }
}
